Added docs for the async voice feature on SwiftUI (#2942)
GetStream/stream-chat-swift: docusaurus/docs/iOS/swiftui/audio-recording.md

---

> title: Voice Recording

Reviewer: What's the difference between `audio-recording.md` and `voice-recording.md`? They even have the same title 🤔

Author: 🤦 fixed now

---

Stream Chat's SwiftUI SDK allows you to record and share async voice messages in your channels. The voice recordings have a built-in attachment type (as defined [here](https://getstream.io/chat/docs/sdk/ios/uikit/guides/working-with-attachments/)).

:::note

> Voice Recordings on SwiftUI are available since version 4.46.0.

Reviewer: What do you think about adding a link to the release on GitHub (I know it's not released yet, but the link will work after releasing)

Author: done

    composerConfig: ComposerConfig(isVoiceRecordingEnabled: true)
)
let streamChat = StreamChat(chatClient: chatClient, utils: utils)
```

## Recording UI Flows

The voice recording feature supports several different UI flows.

### Recording

When the user long presses on the voice recording button longer than 1 second, the recording is started. In that case, while the button is still pressed, the recording view is shown.

The recording view provides the following actions:
- add the recording to the composer input (invoked when releasing the long press button)
- slide to cancel (invoked when you drag to the slide to cancel indicator)
- lock the recording (invoked when you drag towards the lock button)

In order to replace this view with your own implementation, you can implement the following `ViewFactory` method.

```swift
public func makeComposerRecordingView(
        viewModel: MessageComposerViewModel,
        gestureLocation: CGPoint
) -> some View {
    CustomRecordingView(viewModel: MessageComposerViewModel, location: gestureLocation)
```

### Locked View

When the user decides to lock the recording, `LockedView` is presented. It provides the user the following actions:

> - discard the recording - you are back to the initial state.
> - stop the recording - you go into the recording preview state.

Reviewer: Considering the message above, I think `the user` makes more sense, but not a big deal
```suggestion
- discard the recording - the user is back to the initial state.
- stop the recording - the user goes into the recording preview state.
```
